DOCS-6689-made first edits (#171)
mulesoft/docs-api-manager: modules/ROOT/pages/policy-mule4-detokenization.adoc

> [%autowidth.spread,cols="a,a"]

Reviewer: I have been editing this PR for term logic:

Tokenization replaces a value with a token, to protect that value.
Detokenization returns the tokenized value to its original value.

Some of your original text implies just the opposite, so please very carefully evaluate the suggested rewordings to ensure that I have not misunderstood and that your intended meaning is reflected.

Author: Noted and changed accordingly!

|===
>s| Policy Name | Detokenization.

> >s| Summary      | Transforms a tokenized value back to the original data. This policy is designed to work only for Anypoint Runtime Fabric deployments.

Reviewer: ```suggestion
>s| Summary      | Returns a tokenized value to its original value. This policy is designed to work only for Anypoint Runtime Fabric deployments.
```

Author: Done! Thanks!

|===

> Detokenization is the process of detokenizing a token back into its original value to redeem sensitive data for fulfilling specific business needs. For example, an application may need to use a detokenized value to validate a specific credit card number. *<Eva, can we add a better example?>*

Reviewer 2: Tokenization is the process of replacing sensitive information with tokens to minimize the exposure of the original data and reduce the risk of compromising this information. Applications can operate with the tokens instead of the original data and, when there is an exception and the application requires access to the sensitive data, the detokenization policy allows to redeem this data. For example, for business purposes, an application might need to know the bank account number or the credit card number to process it.

Author: @evangelina is this for Tokenization or Detokenization?

Reviewer 2: Context on detokenization and the use case

Author: Done! Thanks!

|===

> Detokenization is the process of transforming tokenized data back into its original value to redeem sensitive data for fulfilling specific business needs. You can tokenization your data to replace sensitive information with tokens to reduce the risk of compromising sensitive information. 

Reviewer: ```suggestion
Detokenization is the process of returning sensitive information that has been tokenized to reduce the risk of compromise back to its original value.
```

Author: Done! Thanks!

> For example, an application can require the bank account number or the credit card number for generating monthly credit card statements. Detokenized sensitive data must always be read under strict security controls.

Reviewer: ```suggestion
For example, an application with the appropriate access permissions can reveal the true values of tokenized bank account numbers or credit card numbers for issuing monthly statements. Detokenized data must always be read under strict security controls.
```

Author: Done! Thanks!

== Prerequisites

To apply and manage the Detokenization policy, you must have:

> * Obtained the *Manage Policies* permission.

Reviewer: ```suggestion
* Obtained the *Manage Policies* permission
```

Author: Done! Thanks!

* To xref:anypoint-security::tokenization-example.adoc[configure and use the tokenization service], you must have:
** Installed Runtime Fabric
** Obtained the *Manage Tokenization Services* permission
* An operating tokenization service

== Scope and Considerations

Before you apply this policy, review the following considerations:

* One instance of the policy supports only one tokenization service. 
+

> All the formats that you configured need to be contained in the same tokenization service.

Reviewer: GLOBAL: need to > must
```suggestion
All the formats that you configured must be contained in the same tokenization service.
```

Author: Done! Thanks!

* You can use one instance of the policy to tokenize a value either in the request or the response, but not in both.

> * The policy is not aware of the state of the tokenization service until runtime. If you deleted your tokenization service or if it is not operational, the policy cannot establish communication with it. This causes all the requests to fail until the service is available again.

Reviewer: ```suggestion
* The policy is not aware of the state of the tokenization service until runtime. 
If you deleted your tokenization service or if it is not operational, the policy cannot establish communication with it. This causes all the requests to fail until the service is available again.
```

Author: Done! Thanks!

* During policy configuration, the services menu displays only the tokenization services that are ready to use. 
+

> If you do not see the service you want, check if it is already deployed.

Reviewer: ```suggestion
If you do not see the service you want, check to see if it is already deployed.
```
Should we add "...by <doing this>" to tell them how?

Author: Thanks!

== How This Policy Works

> The Detokenization policy transforms certain pieces of the payload with the tokenized values. To define which pieces of payload is detokenized, you must provide a selector expression.

Reviewer: ```suggestion
The Detokenization policy returns tokenized information within the payload to its original values. To define which data in the payload is detokenized, you must provide a selector expression.
```

Author: Done! Thanks!

#[payload.socialSecurityNumber]
--

> Now, every time the policy receives a request containing a `socialSecurityNumber` key in its top level, it will extract all the tokens that match the expression you indicated, it will establish a connection with the tokenization service, and finally it will replace the original detokenized values in your payload with the ones provided by the service. 

Reviewer: ```suggestion
Using this selector, every time the policy receives a request containing a `socialSecurityNumber` key in its top level, it extracts all the tokens that match the expression you indicated, establishes a connection with the tokenization service, and replaces the tokenized values in your payload with the original values. 
```
Do we need to include "in its top level"? It seems a useless piece of info to the user, that just complicates the sentence.

Reviewer: Once again, I have been editing this PR for term logic:

Tokenization replaces a value with a token, to protect that value.
Detokenization returns the tokenized value to its original value.

Some of your original text implies just the opposite, so please very carefully evaulate the suggested rewordings to ensure that I have not misunderstood and that your intended meaning is reflected.

Author: Done! Thanks!

}
--

You can configure the detokenization policy to detokenize values in the request or response of your payload.

=== Detokenize Requests

The following events occur when detokenizing requests:

image::policy-mule4-tokenization-67b25.png[]

. A user performs a request to the application protected by the policy.
. The request goes first to Anypoint Security and then to the application Detokenization policy.
. The Detokenization policy extracts all tokens from the payload that match the selector expression.

> . The detokenization service receives the tokens, transforms them back to sensitive data, and returns the result to the policy.

Reviewer: ```suggestion
. The detokenization service receives the tokens, returns them to their original sensitive values, and sends the result to the policy.
```

Author: Done! Thanks!

> . The policy replaces the payload with the detokenized values and forwards it to the Mule application.

Reviewer: ```suggestion
. The policy replaces the tokenized payload with the detokenized values and forwards it to the Mule application.
```

Author: Done! Thanks!

. The Mule application processes the request and returns the result.

=== Detokenizing Responses

The following events occur when detokenizing responses:

image::policy-mule4-tokenization-9347d.png[]

. A user performs a request to the application protected by the policy.
. The request goes first to Anypoint Security and then to the application Detokenization policy.

> . The Detokenization policy tranforms the response of the Mule application by redirecting the request without extracting anything from the payload.

Reviewer: I don't understand this. 
"transforms the response" sounds like the policy changes the values, but the rest of the sentence sounds like the policy does nothing at this point but redirect the request?
Clarify.

Author: Changed it to "The Detokenization policy redirects the request without extracting anything from the payload."
Done! Thanks!

. The Mule application processes the request and returns the result.

> . The detokenization policy extracts the tokens from the payload using the selector expression.

Reviewer: ```suggestion
. The Detokenization policy extracts the tokens from the payload using the selector expression.
```

Author: Done! Thanks!

> . The tokenization service receives the tokens, transforms them back to sensitive data, and  returns the result to the policy.

Reviewer: The tokenization service or the detokenization service?
```suggestion
. The tokenization service receives the tokens, returns them to their original sensitive values, and sends the result to the policy.
```

Author: Done! Thanks!

| No. +
Keep in mind that if you don't specify a selector expression, the policy will be applied, but won't detokenize anything from your payload.

| Format
| Type of information that will be detokenized. Formats may vary depending on the selected service. Choose one format from the list.

If you do not see any formats in the list, check that the tokenization service selected contains at least one format. +
See the xref:anypoint-security::tokenization-formats.adoc[Tokenization Formats] article for more information.
| No. +
If you do not specify a format, the policy will be applied. However, nothing is detokenized from your payload.

|===

== See Also

> 

Reviewer 2: Same comment as with the client id enforcement, we should add the link to the release notes of this policy here.

Author: Added a link! Thanks!
